components/server/src/workspace/workspace-starter.ts

```diff
@@ -683,6 +688,163 @@ export class WorkspaceStarter {
         }
     }
 
+    private async actuallyStartWorkspaceOld(
```

Comment:
This refactoring honestly looks intimidating in the GitHub diff view :confused: 
At it's core it's really not, but hard to proof. I wonder if I should go with the less invasive version just to ease a review... :thinking: 

Let me know what you think!